webapp/src/components/ApiKeyList.tsx

```diff
@@ -0,0 +1,234 @@
+'use strict';
+
+import * as API from '@api';
+import { ArrowPathIcon, PencilSquareIcon, TrashIcon } from '@heroicons/react/20/solid';
+import DeleteModal from 'components/DeleteModal';
+import ErrorAlert from 'components/ErrorAlert';
+import { useAccountContext } from 'context/account';
+import cn from 'lib/cn';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+import { ModelContextWindow, ModelKnowledgeCutoff } from 'struct/model';
+
+import { CopyToClipboardButton } from './chat/message';
+import ConfirmModal from './ConfirmModal';
+
+export default function ApiKeyList({ keys, fetchKeys }: { keys: any[]; fetchKeys?: any }) {
+	const [accountContext]: any = useAccountContext();
+	const { csrf } = accountContext as any;
+	const router = useRouter();
+	const { resourceSlug } = router.query;
+	const [deletingModel, setDeletingModel] = useState(null);
+	const [regeneratingKey, setRegeneratingKey] = useState(null);
+	const [deletingMap, setDeletingMap] = useState({});
+	const [error, setError] = useState(null);
+	const [deleteOpen, setDeleteOpen] = useState(false);
+	const [regenerateOpen, setRegenerateOpen] = useState(false);
+
+	const handleCopyClick = async dataToCopy => {
+		try {
+			await navigator.clipboard.writeText(dataToCopy);
+			toast.success('Copied to clipboard');
+		} catch {
+			/* ignored for now */
+		}
+	};
+	async function deleteKey(keyId, ownerId) {
+		setDeleteOpen(false);
+		API.deleteKey(
+			{
+				_csrf: csrf,
+				ownerId,
+				keyId
+			},
+			() => {
+				fetchKeys();
+				toast('Deleted Key');
+			},
+			() => {
+				toast.error('Error deleting Key');
+			},
+			router
+		);
+	}
+
+	async function regenerateKey(keyId, ownerId) {
+		//TODO: implement regeneration of key
+		API.incrementKeyVersion(
+			{
+				_csrf: csrf,
+				ownerId,
+				keyId
+			},
+			() => {
+				fetchKeys();
+				toast('Regenerated Key');
+			},
+			() => {
+				toast.error('Error Regenerating Key');
+			},
+			router
+		);
+		setRegenerateOpen(false);
```

Comment:
move this and add it after both toasts. Otherwise the Api call is async and this will immediately close the modal